src/test/groovy/com/synacy/gradprogram/spock/exercise/OrderingServiceSpec.groovy

```diff
@@ -0,0 +1,187 @@
+
+    OrderingService orderingService
+    OrderRepository orderRepository = Mock(OrderRepository)
+    def setup(){
+        orderingService = new OrderingService(orderRepository)
+    }
+
+    Item food = new Item("Burger", 15, ItemType.FOOD)
+    Item food2 = new Item("Pizza", 5, ItemType.FOOD)
+    Item gadget = new Item("Cellphone", 10, ItemType.GADGET)
+    Item gadget2 = new Item("iPad", 5, ItemType.GADGET)
+    Item appliance = new Item("TV", 20, ItemType.APPLIANCE)
+    Item appliance2 = new Item("TV", 5, ItemType.APPLIANCE)
+    Item clothing = new Item("Dress", 10, ItemType.CLOTHING)
+    Item clothing2 = new Item("Dress", 5, ItemType.CLOTHING)
+
+    def "cartContainsFoodItem should respond with true if cart contains item with type food"() {
+        given:
+        List<Item> testListItemWFood = [food, gadget, appliance, clothing]
+        def testCart = new Cart(UUID.randomUUID(), testListItemWFood)
+        boolean isContainFood = true
+
+        when:
+        boolean foodItem = orderingService.cartContainsFoodItem(testCart)
+
+        then:
+        isContainFood == foodItem
+    }
+
+    def "cartContainsFoodItem should respond with false if cart does not contains item with type food"() {
+        given:
+        List<Item> testListItemWoFood = [gadget, appliance, clothing]
+        def testCart = new Cart(UUID.randomUUID(), testListItemWoFood)
+        boolean isNotContainFood = false
+        when:
+        boolean foodItem = orderingService.cartContainsFoodItem(testCart)
+
+        then:
+        isNotContainFood == foodItem
+    }
+
+    def "calculateTotalCostOfCart should respond with the total cost of item inside the cart"() {
+        given:
+        List<Item> testListItem = [food, gadget, appliance, clothing]
+        def testCart = new Cart(UUID.randomUUID(), testListItem)
+
+        when:
+        double testTotal = orderingService.calculateTotalCostOfCart(testCart)
+
+        then:
+        55d == testTotal
+    }
+
+    def "isCartEligibleForDiscount should respond with true if total cart price is greater than 50.0 and size is greater than 5"() {
+
+        given:
+        List<Item> testListItem = [food, gadget, appliance, clothing, food2, gadget2]
+        def testCart = new Cart(UUID.randomUUID(), testListItem)
+        boolean isExpectedEligible = true
+
+        when:
+        boolean isEligible = orderingService.isCartEligibleForDiscount(testCart)
+
+        then:
+        isExpectedEligible == isEligible
+    }
+
+    def "isCartEligibleForDiscount should respond with false if cart price is less than 50.0 and cart size is less than 5"() {
+
+        given:
+        List<Item> testListItem = [food, gadget, clothing]
+        def testCart = new Cart(UUID.randomUUID(), testListItem)
+        boolean isNotEligible = false
+
+        when:
+        boolean isEligible = orderingService.isCartEligibleForDiscount(testCart)
+
+        then:
+        isNotEligible == isEligible
```

Comment:
The variable names makes it confusing to read :sweat_smile: Asserting it as this is much better:
```
then:
isEligible
```

This is mostly my fault with my last comment :bow: Let's revert them to the earlier state since the current one is more difficult/confusing to read.